docs/050-breaking-changes.rst

```diff
@@ -0,0 +1,356 @@
+  payable`` is possible via conversion through ``uint160``.
+
+* Conversions between ``bytesX`` and ``uintY`` of different size are now
+  disallowed due to ``bytesX`` padding on the right and ``uintY`` padding on
+  the left which may cause unexpected conversion results.  The size must now be
+  adjusted within the type before the conversion.  For example, you can convert
+  a ``bytes4`` (4 bytes) to a ``uint64`` (8 bytes) by first converting the
+  ``bytes4`` variable to ``bytes8`` and then to ``uint64``.
+
+* Using ``msg.value`` in ``non-payable`` functions (or introducing it via a
+  modifier) is disallowed as a security feature. Turn the function into
+  ``payable`` or create a new internal function for the program logic that
+  uses ``msg.value``.
+
+* For clarity reasons, the command line interface now requires ``-`` if the
+  standard input is used as source.
+
+Deprecated Elements
+===================
+
+This section lists changes that deprecate prior features or syntax.  Note that
+many of these changes were already enabled in the experimental mode
+``v0.5.0``.
+
+* The command line option ``--formal`` was removed. Formal verification is
+  now enabled via ``pragma experimental SMTChecker;``.
+
+* The command line option ``--julia`` was renamed to ``--yul`` due to the
+  renaming of the intermediate language ``Julia`` to ``Yul``.
+
+* Function ``callcode`` is now disallowed (in favor of ``delegatecall``. It
+  is still possible to use it via inline assembly.
+
+* Declaring empty structs is now disallowed for clarity.
+
+* The ``var`` keyword is now disallowed.
+
+* ``suicide`` is now disallowed (in favor of ``selfdestruct``).
+
+* ``sha3`` is now disallowed (in favor of ``keccak256``).
+
+* ``throw`` is now disallowed (in favor of ``revert``, ``require`` and
+  ``assert``).
+
+* The Json AST fields ``constant`` and ``payable`` were removed. The
+  information is now present in the ``stateMutability`` field.
+
+* The ``--clone-bin`` and ``--combined-json clone-bin`` command line options
+  were removed.
+
+* Using ``constant`` as function state mutability modifier is now disallowed.
+
+* Assignments between tuples with different number of components is now
+  disallowed.
+  
+* Values for constants that are not compile-time constants are disallowed.
+
+* New keywords: ``calldata`` and ``constructor``.
+
+* New reserved keywords: ``alias``, ``apply``, ``auto``, ``copyof``,
+  ``define``, ``immutable``, ``implements``, ``macro``, ``mutable``,
+  ``override``, ``partial``, ``promise``, ``reference``, ``sealed``,
+  ``sizeof``, ``supports``, ``typedef`` and ``unchecked``.
+
+* Boolean expressions cannot use arithmetic operations.
+
+* The unit denomination ``years`` is now disallowed.
+
+* Trailing dots that are not followed by a number are now disallowed.
+
+* Remappings with empty prefix are disallowed.
+
+* Combining hex numbers with unit denominations (e.g. ``0x1e wei``) is now
```

Comment:
I think we should group some of them:

 - literals / suffixes
 - commandline interface
 - constructors
 - variables
 - conversions
 - syntax